storageinsights/api/StorageInsights.Samples/GetInventoryReportNames.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START storageinsights_get_inventory_report_names]
+using Google.Cloud.StorageInsights.V1;
+using System;
+using System.Collections.Generic;
+
+namespace StorageInsights.Samples
+{
+    public class GetInventoryReportNamesSample
+    {
+        public IEnumerable<string> GetInventoryReportNames(
+            string projectId = "your-project-id",
+            string bucketLocation = "us-west-1",
+            string inventoryReportConfigUuid = "2b90d21c-f2f4-40b5-9519-e29a78f2b09f")
+        {
+            StorageInsightsClient storageInsightsClient = StorageInsightsClient.Create();
+            ReportConfigName reportConfigName =
+                new ReportConfigName(projectId, bucketLocation, inventoryReportConfigUuid);
+            ReportConfig reportConfig = storageInsightsClient.GetReportConfig(reportConfigName);
+            string extension = reportConfig.CsvOptions == null ? "csv" : "parquet";
+            var reportNames = new List<string>();
+
+            Console.WriteLine("You can use the Google Cloud Storage Client to download the following objects from " +
+                              "Google Cloud Storage:");
+            foreach(ReportDetail reportDetail in storageInsightsClient.ListReportDetails(reportConfigName))
+            {
+                for (long index = reportDetail.ShardsCount - 1; index >= 0; index--)
+                {
+                    String reportName = reportDetail.ReportPathPrefix + index + "." + extension;
+                    reportNames.Add(reportName);
```

Comment:
This isn't just for testing, this is what  a user would have to do get the list of files to query GCS for

Reply:
Oh I meant adding the calculate `reportName`s to a list to return that list. Yes of course you have to show how to calculate the report names, and then you print them in the console for demonstration purposes. What's odd is to also add them to a list. Some users would want them on a list, some users will write them to a file, some users will inmediately start processes to download the reports etc... So, I just meant stop building and returning the list with all the reports in it. Adn you are not using this for testing so that's fine.